src/Orleans/Async/Cancellation/GrainCancellationToken.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.CodeGeneration;
+using Orleans.Runtime;
+using Orleans.Serialization;
+
+namespace Orleans.Async
+{
+    /// <summary>
+    /// Grain cancellation token that can be passed thought cross-domain boundaries
+    /// </summary>
+    [Serializable]
```

Comment:
it might be fine, although I'm not sure what's the scenario where it could cross domains outside of Orleans during a grain call (and hence the need to support the .NET binary formatter). Is this required for unit testing in different AppDomains or what?